pkgs/development/compilers/cudatoolkit/redist/overrides.nix

```diff
@@ -52,34 +52,8 @@ in
   );
 
   cuda_nvcc = prev.cuda_nvcc.overrideAttrs (_: {
-    # Required by cmake's enable_language(CUDA) to build a test program
-    # When implementing cross-compilation support: this is
-    # final.pkgs.targetPackages.cudaPackages.cuda_cudart
-    env = {
-      # Given the multiple-outputs each CUDA redist has, we can specify the exact components we
-      # need from the package. CMake requires:
-      # - the cuda_runtime.h header, which is in the dev output
-      # - the dynamic library, which is in the lib output
-      # - the static library, which is in the static output
-      cudartInclude = "${final.cuda_cudart.dev}";
-      cudartLib = "${final.cuda_cudart.lib}";
-      cudartStatic = "${final.cuda_cudart.static}";
-    };
-
-    # Point NVCC at a compatible compiler
-
-    # Desiredata: whenever a package (e.g. magma) adds cuda_nvcc to
-    # nativeBuildInputs (offsets `(-1, 0)`), magma should also source the
-    # setupCudaHook, i.e. we want it the hook to be propagated into the
-    # same nativeBuildInputs.
-    #
-    # Logically, cuda_nvcc should include the hook in depsHostHostPropagated,
-    # so that the final offsets for the propagated hook would be `(-1, 0) +
-    # (0, 0) = (-1, 0)`.
-    #
-    # In practice, TargetTarget appears to work:
-    # https://gist.github.com/fd80ff142cd25e64603618a3700e7f82
-    depsTargetTargetPropagated = [
+    # One'd expect this should be depsHostHostPropagated, but that doesn't work
+    propagatedBuildInputs = [
```

Comment:
Here's evidence that `propagatedHostTargetDeps` (propagatedBuildInputs) work and `HostHost` doesn't: https://gist.github.com/SomeoneSerge/7d0e633743175bee6470b03281fe74e1

Same goes for `BuildHost` instead of `BuildBuild` when propagating the hook to nvcc's users. Whatever the reason, a transitive dependency is assigned the offsets of `(prevHost + relHost, prevHost + relTarget)` rather than `(prevHost + relHost, prevTarget + relTarget)`: https://github.com/NixOS/nixpkgs/blob/82fa717f858db423d0275dfe028352424b29fce4/pkgs/stdenv/generic/setup.sh#L552-L560

The relevant message in the cross-compilation matrix: https://matrix.to/#/%23cross-compiling%3Anixos.org/%248lsLV15zqjt5JdopUy-3aD3o8ym1NdITs18bZRZXMc8?via=someonex.net&via=matrix.org&via=catgirl.cloud&via=nixos.dev (yet to receive replies)